corehq/apps/domain/tests/test_utils.py

```diff
@@ -32,3 +34,17 @@ def test_guess_domain_language(self):
     def test_guess_domain_language_no_apps(self):
         lang = guess_domain_language(self.domain_name)
         self.assertEqual('en', lang)
+
+
+@contextmanager
+def test_domain(skip_full_delete=False):
+    """Context manager for use in tests"""
+    from corehq.apps.domain.shortcuts import create_domain
+    domain = create_domain('domain')
+    try:
+        yield domain
+    finally:
+        if skip_full_delete:
+            Domain.get_db().delete_doc(domain.get_id)
```

Comment:
How do you know whether or not it's safe to only delete the doc? I'm thinking less about this PR and more about all the other tests where we do the full `domain.delete()`.

Reply:
If you have created other couch docs that you aren't cleaning up manually then you'll need to do a full delete. In most cases there are only a few docs that are being created by tests and it would be much faster to delete them manually than to rely on `domain.delete`